modular_skyrat/modules/modular_implants/code/nif_presistance.dm

```diff
@@ -0,0 +1,87 @@
+/// How much damage is done to the NIF if the user ends the round with it uninstalled?
+#define LOSS_WITH_NIF_UNINSTALLED 25
+
+///Saves the NIF data for everyone inside of the server.
+/datum/controller/subsystem/persistence/proc/save_nifs()
+	for(var/player in GLOB.joined_player_list)
+		var/mob/living/carbon/human/ending_human = get_mob_by_ckey(player)
+		if(!istype(ending_human) || !ending_human.mind?.original_character_slot_index)
+			continue
+
+		var/mob/living/carbon/human/original_human = ending_human.mind.original_character.resolve()
+
+		if(!original_human)
+			continue
+
+		original_human.save_nif_data()
+
+///Saves the NIF data for a individual user.
+/mob/living/carbon/human/proc/save_nif_data()
+	if(!ckey || !mind?.original_character_slot_index)
+		return
+
+	var/path = "data/player_saves/[ckey[1]]/[ckey]/nif.sav"
+	var/savefile/save = new /savefile(path)
+	var/obj/item/organ/internal/cyberimp/brain/nif/saved_nif
+
+	if(HAS_TRAIT(src, GHOSTROLE_TRAIT)) //Nothing is lost from playing a ghost role
+		return FALSE
+
+	if(installed_nif)
+		if(installed_nif.durability == 0)
```

Comment:
can durability go negative?

Reply:
Nope, but let me change that anyways just incase that ends up being an issue